src/app/animals/animals.component.ts

```diff
@@ -0,0 +1,72 @@
+import { AnimalService } from '../shared/services/animal.service';
+import { Animal } from '../animal';
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-animals',
+  templateUrl: './animals.component.html',
+  styleUrls: ['./animals.component.css'],
+})
+export class AnimalsComponent implements OnInit {
+  animals: Animal[] = [];
+
+  constructor(private animalService: AnimalService) {}
+
+  searchName: string;
+  searchId: number;
+  searchState: boolean;
+  adoption: boolean = false;
+
+  ngOnInit(): void {
+    this.getAnimal();
+  }
+
+  getAdopted(): void {
+    this.animalService
+      .getAnimals()
+      .subscribe(
+        (animals) =>
+          (this.animals = animals.filter((Animal) => Animal.adoption === true))
+      );
+  }
+
+  getNotAdopted(): void {
```

Comment:
a`getAdopted` e `getNotAdopted` sono praticamente uguali a parte la condizione. Dovresti trovare un modo per astrarre la logica in un metodo solo magari passando un parametro.